promptsource/promptsource.py

```diff
@@ -103,11 +105,11 @@ def show_text(t, width=WIDTH):
     all_infos = manager.dict()
     all_datasets = list(set([t[0] for t in template_collection.keys]))
 
-    def get_infos(d_name):
+    def get_infos(d_name, all_infos):
```

Comment:
This isn't mandatory "here" but better be. Normally a function must be defined before Pool init. However, in this case it is almost impossible, because the Pool context is managed by streamlit. Assuming we actually can control the context, then the Pool will encounter `AttributeError` for the implicit/global variable `all_infos`.